pact/brackets-review.pact

```diff
@@ -0,0 +1,504 @@
+  )
+
+  (defcap BRACKETS-GOVERNANCE ()
+    (enforce-brackets-admin)
+  )
+
+  (use coin)
+
+  (defconst CONTRACT_ACCOUNT:string 'master-bracket-keyset)
+  (defun contract-guard:guard () (create-module-guard 'master-bracket-keyset))
+
+   ;game initiated, people still signing up
+   (defconst INITIATED:string "initiated")
+   ;people signed up, game in progress
+   (defconst IN_PROGRESS:string "in-progress")
+   ;game finished and winner not paid
+   (defconst COMPLETE:string "complete")
+   ;game finished and winner paid
+   (defconst WINNER_PAID:string "winner-paid")
+   ;team has no player assigned
+   (defconst UNASSIGNED:string "unassigned")
+
+  (defschema empty-bracket
+    ;will need to refactor admin to take in a keyset
+    ;not sure how to do this from the frontend though...
+    admin:string
+    bracket:list
+    status:string
+    moneyPool:integer
+    players:[string]
+    winner:string
+    entryFee:integer
+    usernames:[string])
+    ;guard:guard
+
+  (deftable empty-bracket-table:{empty-bracket})
+
+  (defschema bracket-betting
+    ;will need to refactor admin to take in a keyset
+    ;not sure how to do this from the frontend though...
+    admin:string
+    bracket:list
+    status:string
+    moneyPool:integer
+    players:[string]
+    players-bets:list
+    winner:string
+    entryFee:integer)
+
+  (deftable bracket-betting-table:{bracket-betting})
+
+  (defschema users
+    username:string
+    user-guard:guard
+    balance:integer
+    bb-games:list
+    bb-admins:list
+    eb-games:list
+    eb-admins:list
+    games-won:list)
+
+  (deftable users-table:{users})
+
+  (defcap BRACKET-ADMIN-BB (admin-key:string bracket-name:string)
+    (with-read bracket-betting-table bracket-name { "admin":= admin-key-db }
+      (enforce (= admin-key admin-key-db) "you are not the bracket admin")
+    )
+  )
+
+  (defcap BRACKET-ADMIN-EB (admin-key:string bracket-name:string)
+    (with-read empty-bracket-table bracket-name { "admin":= admin-key-db }
+      (enforce (= admin-key admin-key-db) "you are not the bracket admin")
+    )
+  )
+
+  (defcap USER-CAN-TRANSFER (from:string amount:integer)
+    (with-read users-table from { "balance" := from-bal }
+      (enforce (>= from-bal amount) "Insufficient Funds")
+    )
+  )
+
+  (defcap EB-CONTRACT-CAN-TRANSFER (bracket-name:string amount:integer)
+    (with-read empty-bracket-table bracket-name { "moneyPool" := from-bal }
+      (enforce (>= from-bal amount) "Insufficient Funds")
+    )
+  )
+
+  (defcap BB-CONTRACT-CAN-TRANSFER (bracket-name:string amount:integer)
+    (with-read bracket-betting-table bracket-name { "moneyPool" := from-bal }
+      (enforce (>= from-bal amount) "Insufficient Funds")
+    )
+  )
+
+  (defcap IS-REGISTERED-USER (user-key:string)
+    (let ((users (keys users-table)))
+          ;should be user-id
+        (enforce (contains user-key users) "you are not a registered user")
+    )
+  )
+
+  ;needs to be passed as arg to function (the guard)
```

Comment:
Instead of writing comments above a function, the lisp way is to simply document the function:
```
(defun init-user (...)
   "Initialize user with id USER-KEY and name USERNAME. \
    \ Reads user keyset from payload keyed by USER-KEY"
   (insert users-table user-key { 
   ...
```